File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        # print(n)
        while((self.solve(n)%t )!=0):
            n=n+1
        return n
    def solve(self,n):
        ans=1
        while(n>0):
            ans=ans*(n%10)
            if(ans==0):
                return 0
            n=n//10
        print(ans)
        return ans

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.java
class Solution {
    public int smallestNumber(int n, int t) {
    while(solve(n)%t!=0){
        n=n+1;
    }  
    return n;
    }
    int solve(int n){
    
        int ans=1;
        while(n>0){
            ans=ans*(n%10);
            n=n/10;
            if(ans==0){
                return ans;
            }
        }
        return ans;
    }
}